ui/models/job.js

```diff
@@ -115,37 +116,43 @@ export default class JobModel {
     return JobModel.getList(repoName, options, config);
   }
 
-  static async retrigger(jobIds, repoName, getGeckoDecisionTaskId, notify) {
-    const jobTerm = jobIds.length > 1 ? 'jobs' : 'job';
+  static async retrigger(jobs, repoName, notify) {
+    const jobTerm = jobs.length > 1 ? 'jobs' : 'job';
 
     try {
       notify(`Attempting to retrigger/add ${jobTerm} via actions.json`, 'info');
 
-      const jobs = await JobModel.getList(repoName, { id__in: jobIds.join() });
+      const pushIds = [...new Set(jobs.map(job => job.push_id))];
+      const resp = await PushModel.getDecisionTaskMap(pushIds);
+      const taskIdMap = resp.data;
       const uniquePerPushJobs = groupBy(jobs, job => job.push_id);
 
       for (const [key, value] of Object.entries(uniquePerPushJobs)) {
-        getGeckoDecisionTaskId(Number(key), repoName).then(decisionTaskId => {
-          TaskclusterModel.load(decisionTaskId).then(async results => {
-            const actionTaskId = slugid();
-            const addNewJobsTask = results.actions.find(
-              action => action.name === 'add-new-jobs',
-            );
+        const decisionTaskId = taskIdMap[key];
 
-            await TaskclusterModel.submit({
-              action: addNewJobsTask,
-              actionTaskId,
-              decisionTaskId,
-              taskId: null,
-              task: null,
-              input: { tasks: value.map(job => job.job_type_name) },
-              staticActionVariables: results.staticActionVariables,
-            }).then(() =>
+        TaskclusterModel.load(decisionTaskId).then(async results => {
+          const actionTaskId = slugid();
+          const addNewJobsTask = results.actions.find(
+            action => action.name === 'add-new-jobs',
+          );
+
+          await TaskclusterModel.submit({
+            action: addNewJobsTask,
+            actionTaskId,
+            decisionTaskId,
+            taskId: null,
+            task: null,
+            input: { tasks: value.map(job => job.job_type_name) },
+            staticActionVariables: results.staticActionVariables,
+          })
+            .then(() =>
               notify(
                 `Request sent to retrigger/add new jobs via actions.json (${actionTaskId})`,
               ),
-            );
-          });
+            )
+            .catch(error => {
```

Comment:
Does this this catch`4xx` errors?

Reply:
This ``catch`` will throw up an error for 4XX, yeah.  Like if you don't have the scopes to retrigger (like I don't) then you need this catch to tell the user why it failed.